### PR TITLE
Rename `baremetal_0` to `t1.small.x86`

### LINKED ISSUE #514: Rename `baremetal_0` to `t1.small.x86`
In Packet docs and some places we use the term `baremetal_0` instead of `t1.small.x86`.

`baremetal_0` is not a well known term and a user can't look up for that on the internet.

### PR DESCRIPTION
**Pull request by knrt10**
baremetal_0 is not a well known term and a user can't look up for that on the internet. Fixes #514

docs/*: fixed it to `t1.small.x86`
assets/*: fixed it to `t1.small.x86`

Also I previously used `go1.13.5` and didn't new `go1.14` was needed so added that in contribution guidelines

Signed-off-by: knrt10 <kautilya@kinvolk.io>

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ git clone git@github.com:kinvolk/lokomotive.git
 cd lokomotive
 ```
 
+## Requirements
+
+- Minimum Go version **1.4**
+- [golangci-lint](https://github.com/golangci/golangci-lint) installed locally
+
```
Comment (surajssd) [inline docs/CONTRIBUTING.md:14]: Can you add this to a separate commit? Since it is an unrelated change?

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ git clone git@github.com:kinvolk/lokomotive.git
 cd lokomotive
 ```
 
+## Requirements
+
+- Minimum Go version **1.4**
+- [golangci-lint](https://github.com/golangci/golangci-lint) installed locally
```
Comment (invidian) [inline docs/CONTRIBUTING.md:13]: This is not really a contributing requirement (`make` or `make build` does not require it). The CI will run linter anyway. But it's definitely a nice to have to run linter locally before submitting the PR.

Maybe we could add another section, for example "Testing and linting"?

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -7,6 +7,11 @@ git clone git@github.com:kinvolk/lokomotive.git
 cd lokomotive
 ```
 
+## Requirements
+
+- Minimum Go version **1.4**
```
Comment (invidian) [inline docs/CONTRIBUTING.md:12]: It seems to me, that minimal version of Go we need is 1.12, as building with 1.11 gives following compile errors:
```
CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build \
        -mod=vendor \
        -ldflags "-X github.com/kinvolk/lokomotive/pkg/version.Version=`git describe --tags --always`-dirty -extldflags '-static'" \
        -buildmode=exe \
        -o lokoctl \
        github.com/kinvolk/lokomotive/cmd/lokoctl
# github.com/docker/cli/cli/config
vendor/github.com/docker/cli/cli/config/config.go:109:18: undefined: os.UserHomeDir
# helm.sh/helm/v3/pkg/chart/loader
vendor/helm.sh/helm/v3/pkg/chart/loader/archive.go:145:7: undefined: strings.ReplaceAll
# helm.sh/helm/v3/pkg/kube
vendor/helm.sh/helm/v3/pkg/kube/client.go:525:21: undefined: strings.ReplaceAll
make: *** [Makefile:52: build-slim] Error 2
make: *** [Makefile:35: build-in-docker] Error 2
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -7,6 +7,11 @@ git clone git@github.com:kinvolk/lokomotive.git
 cd lokomotive
 ```
 
+## Requirements
```
Comment (invidian) [inline docs/CONTRIBUTING.md:10]: I think it would be good to add this in separate commit, so the additions are logically separated, which allows to reference and revert them independently, if needed in the future.

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (knrt10) [reply docs/CONTRIBUTING.md:14]: Sure

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (iaguis) [inline docs/CONTRIBUTING.md:12]: Do you mean 1.14?

Comment (knrt10) [reply docs/CONTRIBUTING.md:12]: Oh yes. Will update it in next commit as Suraj said

File: docs/configuration-reference/components/cluster-autoscaler.md [hunk 5]
```diff
@@ -83,7 +83,7 @@ Example:
 | `provider`                   | Supported provider, currently Packet.                                                    | "packet"      | false    |
 | `packet.project_id`          | Packet Project ID where the cluster is running.                                          | -             | true     |
 | `packet.facility`            | Packet Facility where the cluster is running.                                            | -             | true     |
-| `packet.worker_type`         | Machine type for workers spawned by the Cluster Autoscaler.                              | "baremetal_0" | false    |
```
Comment (invidian) [inline docs/configuration-reference/components/cluster-autoscaler.md:86]: Can we please align table columns? New default value is longer than the previous one. They then look nicer while editing in plain editor.

**Comment by ipochi** [comment]
I dont know why but for Packet `baremetal_0` is just another reference to `t1.small.x86`

curl -H 'X-Auth-Token: <YOUR_API_TOKEN>' https://api.packet.net/plans | jq | grep "baremetal_0" -A 10
```
"slug": "baremetal_0",
      "name": "t1.small.x86",
      "description": "Our Type 0 configuration is a general use \"cloud killer\" server, with a Intel Atom 2.4Ghz processor and 8GB of RAM.",
      "line": "baremetal",
      "specs": {
        "cpus": [
          {
            "count": 1,
            "type": "Intel Atom C2550 @ 2.4Ghz"
          }
        ],
```

File: docs/configuration-reference/components/cluster-autoscaler.md (hunk 5 above)
Comment (ipochi) [reply docs/configuration-reference/components/cluster-autoscaler.md:86]: use this to format http://markdowntable.com/

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (knrt10) [reply docs/CONTRIBUTING.md:12]: @invidian I tried with `go1.13.5` and method `t.Cleanup` was undefined. I got to find it from [here](https://ieftimov.com/post/testing-in-go-clean-tests-using-t-cleanup/)

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (knrt10) [reply docs/CONTRIBUTING.md:13]: Yea testing section sounds great

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (invidian) [reply docs/CONTRIBUTING.md:12]: Ah yes, hehe. I didn't run the tests :) I'm all in favor for requiring 1.14 then :+1:

File: docs/CONTRIBUTING.md [hunk 6]
```diff
@@ -1,5 +1,16 @@
 # Contribution Guidelines
 
+## Contents
+
+- [Setup developer environment](#setup-developer-environment)
+- [Build the code](#build-the-code)
+- [Build with docker](#build-with-docker)
+- [Update assets](#update-assets)
+- [Authoring PRs](#authoring-prs)
+    - [Commit Format](#commit-format)
```
Comment (surajssd) [inline docs/CONTRIBUTING.md:10]: nit: this has two extra space of indentation.

Comment (knrt10) [reply docs/CONTRIBUTING.md:10]: Fixing it
